dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/net/TracingRequestInterceptorTest.kt

```diff
@@ -133,22 +134,30 @@ internal class TracingRequestInterceptorTest {
     lateinit var fakePackageVersion: String
 
     @BeforeEach
-    fun `set up`() {
+    fun `set up`(forge: Forge) {
+        fakeWhitelistedDomain = forge.aStringMatching("[a-z0-9_]{8}\\.[a-z]{3}")
```

Comment:
The [RFC 3986](https://tools.ietf.org/html/rfc3986) gives the following definition of what a host name can be
```
reg-name      = *( unreserved / pct-encoded / sub-delims )
unreserved    = ALPHA / DIGIT / "-" / "." / "_" / "~"
sub-delims    = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="
pct-encoded   = "%" HEXDIG HEXDIG // e.g.: %20
```

You're only testing a trivial subset, it'd be nice to be sure the code works for any valid host. Especially since you're generating a regex that could embed special characters.